filebeat/beater/crawler.go

```diff
@@ -107,34 +104,41 @@ func (c *crawler) Start(
 		}()
 	}
 
-	logp.Info("Loading and starting Inputs completed. Enabled inputs: %v", len(c.inputs))
+	log.Infof("Loading and starting Inputs completed. Enabled inputs: %v", len(c.inputs))
 
 	return nil
 }
 
 func (c *crawler) startInput(
 	pipeline beat.Pipeline,
 	config *common.Config,
-	states []file.State,
 ) error {
 	if !config.Enabled() {
 		return nil
 	}
 
-	connector := c.out(pipeline)
-	p, err := input.New(config, connector, c.beatDone, states, nil)
+	var h map[string]interface{}
+	config.Unpack(&h)
+	id, err := hashstructure.Hash(h, nil)
 	if err != nil {
-		return fmt.Errorf("Error while initializing input: %s", err)
+		return fmt.Errorf("can not compute id from configuration: %v", err)
+	}
+	if _, ok := c.inputs[id]; ok {
+		return fmt.Errorf("input with same ID already exists: %v", id)
 	}
-	p.Once = c.once
 
-	if _, ok := c.inputs[p.ID]; ok {
-		return fmt.Errorf("Input with same ID already exists: %d", p.ID)
+	runner, err := c.inputsFactory.Create(pipeline, config, nil)
+	if err != nil {
+		return fmt.Errorf("Error while initializing input: %+v", err)
+	}
+	if inputRunner, ok := runner.(*input.Runner); ok {
+		inputRunner.Once = c.once
```

Comment:
`input.New` returns a partially initialized type. The `once` flag must only be set for static inputs, but the factory is used for static and dynamic inputs. As I expect the crawler to go away in the future I wanted keep the scope rather small for what I need for the integration of the new input API, I still keep this small piece of tech debt here for now.